images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1

```diff
@@ -15,279 +17,216 @@ Import-Module (Join-Path $PSScriptRoot "SoftwareReport.Java.psm1") -DisableNameC
-    (Get-DacFxVersion),
-    (Get-MySQLVersion),
-    (Get-SQLPSVersion),
-    (Get-SQLOLEDBDriverVersion)
-)
-
-$markdown += New-MDList -Style Unordered -Lines ($databaseTools | Sort-Object)
-
-$markdown += Build-WebServersSection
-
-$vs = Get-VisualStudioVersion
-$markdown += New-MDHeader "$($vs.Name)" -Level 3
-$markdown += $vs | New-MDTable
-$markdown += New-MDNewLine
-
-$markdown += New-MDHeader "Workloads, components and extensions:" -Level 4
-$markdown += ((Get-VisualStudioComponents) + (Get-VisualStudioExtensions)) | New-MDTable
-$markdown += New-MDNewLine
-
-$markdown += New-MDHeader "Microsoft Visual C++:" -Level 4
-$markdown += Get-VisualCPPComponents | New-MDTable
-$markdown += New-MDNewLine
-
-$markdown += New-MDHeader "Installed Windows SDKs" -Level 4
-$sdk = Get-WindowsSDKs
-$markdown += New-MDNewLine
-$markdown += New-MDList -Lines $sdk.Versions -Style Unordered
-
-$markdown += New-MDHeader ".NET Core Tools" -Level 3
-$dotnetSdk = Get-DotnetSdks
-$dotnetFrameworkVersions = Get-DotnetFrameworkVersions
-$dotnetTools = @(
-    ".NET Core SDK: $($dotnetSdk.Versions)",
-    ".NET Framework: $($dotnetFrameworkVersions)"
-)
-$dotnetTools += Get-DotnetRuntimes | ForEach-Object {
-    "$($_.Runtime): $($_.Versions)"
-}
-$dotnetTools += Get-DotnetTools
-$markdown += New-MDList -Style Unordered -Lines $dotnetTools
+# Cached Tools
+$installedSoftware.AddHeader("Cached Tools").AddNodes($(Build-CachedToolsSection))
+
+# Databases
+$databases = $installedSoftware.AddHeader("Databases")
+$databases.AddHeader("PostgreSQL").AddTable($(Get-PostgreSQLTable))
+$databases.AddHeader("MongoDB").AddTable($(Get-MongoDBTable))
+
+# Database tools
+$databaseTools = $installedSoftware.AddHeader("Database tools")
+$databaseTools.AddToolVersion("Azure CosmosDb Emulator", $(Get-AzCosmosDBEmulatorVersion))
+$databaseTools.AddToolVersion("DacFx", $(Get-DacFxVersion))
+$databaseTools.AddToolVersion("MySQL", $(Get-MySQLVersion))
+$databaseTools.AddToolVersion("SQL OLEDB Driver", $(Get-SQLOLEDBDriverVersion))
+$databaseTools.AddToolVersion("SQLPS", $(Get-SQLPSVersion))
+
+# Web Servers
+$installedSoftware.AddHeader("Web Servers").AddTable($(Build-WebServersSection))
+
+# Visual Studio
+$vsTable = Get-VisualStudioVersion
+$visualStudio = $installedSoftware.AddHeader($vsTable.Name)
+$visualStudio.AddTable($vsTable)
+
+$workloads = $visualStudio.AddHeader("Workloads, components and extensions")
+$workloads.AddTable((Get-VisualStudioComponents) + (Get-VisualStudioExtensions))
 
+$msVisualCpp = $visualStudio.AddHeader("Microsoft Visual C++")
+$msVisualCpp.AddTable($(Get-VisualCPPComponents))
+
+$visualStudio.AddToolVersionsList("Installed Windows SDKs", $(Get-WindowsSDKs).Versions, '^.+')
+
+# .NET Core Tools
+$netCoreTools = $installedSoftware.AddHeader(".NET Core Tools")
+$netCoreTools.AddToolVersionsListInline(".NET Core SDK", $(Get-DotnetSdks).Versions, '^\d+\.\d+\.\d')
+$netCoreTools.AddToolVersionsListInline(".NET Framework", $(Get-DotnetFrameworkVersions), '^.+')
```

Comment:
A bit better regex

```suggestion
$netCoreTools.AddToolVersionsListInline(".NET Framework", $(Get-DotnetFrameworkVersions), '^\d+\.\d+')
```

Reply:
here not changed and left the previous one regex string, as it seems like this way the versions 4.8 and 4.8.1 would have a major ver collision